Release Zebra v1.2.0

oxarbitrage opened the pull request:
---
name: 'Release Checklist Template'
about: 'Checklist to create and publish a Zebra release'
title: 'Release Zebra (version)'
labels: 'A-release, C-trivial, P-Critical :ambulance:'
assignees: ''

---

# Prepare for the Release

- [x] Make sure the PRs with the new checkpoint hashes and missed dependencies are already merged.
      (See the release ticket checklist for details)


# Summarise Release Changes

These steps can be done a few days before the release, in the same PR:

## Change Log

**Important**: Any merge into `main` deletes any edits to the draft changelog.
Once you are ready to tag a release, copy the draft changelog into `CHANGELOG.md`.

We use [the Release Drafter workflow](https://github.com/marketplace/actions/release-drafter) to automatically create a [draft changelog](https://github.com/ZcashFoundation/zebra/releases). We follow the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.

To create the final change log:
- [x] Copy the **latest** draft changelog into `CHANGELOG.md` (there can be multiple draft releases)
- [x] Delete any trivial changes
    - [x] Put the list of deleted changelog entries in a PR comment to make reviewing easier
- [x] Combine duplicate changes
- [x] Edit change descriptions so they will make sense to Zebra users
- [x] Check the category for each change
  - Prefer the "Fix" category if you're not sure

## README

README updates can be skipped for urgent releases.

Update the README to:
- [x] Remove any "Known Issues" that have been fixed since the last release.
- [x] Update the "Build and Run Instructions" with any new dependencies.
      Check for changes in the `Dockerfile` since the last tag: `git diff <previous-release-tag> docker/Dockerfile`.
- [x] If Zebra has started using newer Rust language features or standard library APIs, update the known working Rust version in the README, book, and `Cargo.toml`s

You can use a command like:
```sh
fastmod --fixed-strings '1.58' '1.65'
```

## Create the Release PR

- [x] Push the updated changelog and README into a new branch
      for example: `bump-v1.0.0` - this needs to be different to the tag name
- [x] Create a release PR by adding `&template=release-checklist.md` to the comparing url ([Example](https://github.com/ZcashFoundation/zebra/compare/bump-v1.0.0?expand=1&template=release-checklist.md)).
- [x] Freeze the [`batched` queue](https://dashboard.mergify.com/github/ZcashFoundation/repo/zebra/queues) using Mergify.
- [x] Mark all the release PRs as `Critical` priority, so they go in the `urgent` Mergify queue.
- [x] Mark all non-release PRs with `do-not-merge`, because Mergify checks approved PRs against every commit, even when a queue is frozen.


# Update Versions and End of Support

## Update Zebra Version

### Choose a Release Level

Zebra follows [semantic versioning](https://semver.org). Semantic versions look like: MAJOR.MINOR.PATCH[-TAG.PRE-RELEASE]

Choose a release level for `zebrad`. Release levels are based on user-visible changes from the changelog:
- Mainnet Network Upgrades are `major` releases
- significant new features or behaviour changes; changes to RPCs, command-line, or configs; and deprecations or removals are `minor` releases
- otherwise, it is a `patch` release

Zebra's Rust API doesn't have any support or stability guarantees, so we keep all the `zebra-*` and `tower-*` crates on a beta `pre-release` version.

### Update Crate Versions

<details>

<summary>If you're publishing crates for the first time, click this triangle for extra steps</summary>

- [x] Install `cargo-release`: `cargo install cargo-release`
- [x] Make sure you are  an owner of the crate or [a member of the Zebra crates.io `owners` group on GitHub](https://github.com/orgs/ZcashFoundation/teams/owners)

</details>

Check that the release will work:
- [x] Update crate versions, commit the changes to the release branch, and do a release dry-run:

```sh
cargo release version --verbose --execute --allow-branch '*' --workspace --exclude zebrad beta
cargo release version --verbose --execute --allow-branch '*' --package zebrad patch # [ major | minor | patch ]
cargo release replace --verbose --execute --allow-branch '*' --package zebrad
cargo release commit --verbose --execute --allow-branch '*'
```

Crate publishing is [automatically checked in CI](https://github.com/ZcashFoundation/zebra/actions/workflows/release-crates-io.yml) using "dry run" mode.

## Update End of Support

The end of support height is calculated from the current blockchain height:
- [x] Find where the Zcash blockchain tip is now by using a [Zcash explorer](https://zcashblockexplorer.com/blocks) or other tool.
- [x] Replace `ESTIMATED_RELEASE_HEIGHT` in [`end_of_support.rs`](https://github.com/ZcashFoundation/zebra/blob/main/zebrad/src/components/sync/end_of_support.rs) with the height you estimate the release will be tagged.

<details>

<summary>Optional: calculate the release tagging height</summary>

- Add `1152` blocks for each day until the release
- For example, if the release is in 3 days, add `1152 * 3` to the current Mainnet block height

</details>

## Update the Release PR

- [x] Push the version increments and the release constants to the release branch.


# Publish the Zebra Release

## Create the GitHub Pre-Release

- [x] Wait for all the release PRs to be merged
- [x] Create a new release using the draft release as a base, by clicking the Edit icon in the [draft release](https://github.com/ZcashFoundation/zebra/releases)
- [x] Set the tag name to the version tag,
      for example: `v1.0.0`
- [x] Set the release to target the `main` branch
- [x] Set the release title to `Zebra ` followed by the version tag,
      for example: `Zebra 1.0.0`
- [x] Replace the prepopulated draft changelog in the release description with the final changelog you created;
      starting just _after_ the title `## [Zebra ...` of the current version being released,
      and ending just _before_ the title of the previous release.
- [x] Mark the release as 'pre-release', until it has been built and tested
- [x] Publish the pre-release to GitHub using "Publish Release"
- [x] Delete all the [draft releases from the list of releases](https://github.com/ZcashFoundation/zebra/releases)

## Test the Pre-Release

- [x] Wait until the [Docker binaries have been built on `main`](https://github.com/ZcashFoundation/zebra/actions/workflows/continous-integration-docker.yml), and the quick tests have passed.
- [x] Wait until the [pre-release deployment machines have successfully launched](https://github.com/ZcashFoundation/zebra/actions/workflows/continous-delivery.yml)

## Publish Release

- [x] [Publish the release to GitHub](https://github.com/ZcashFoundation/zebra/releases) by disabling 'pre-release', then clicking "Set as the latest release"

## Publish Crates

- [x] Run `cargo login`
- [x] Run `cargo clean` in the zebra repo (optional)
- [x] Publish the crates to crates.io: `cargo release publish --verbose --workspace --execute`
- [x] Check that Zebra can be installed from `crates.io`:
      `cargo install --locked --force --version 1.minor.patch zebrad && ~/.cargo/bin/zebrad`
      and put the output in a comment on the PR.

## Publish Docker Images
- [x] Wait for the [the Docker images to be published successfully](https://github.com/ZcashFoundation/zebra/actions/workflows/release-binaries.yml).
- [x] Un-freeze the [`batched` queue](https://dashboard.mergify.com/github/ZcashFoundation/repo/zebra/queues) using Mergify.
- [x] Remove `do-not-merge` from the PRs you added it to

## Release Failures

If building or running fails after tagging:

<details>

<summary>Tag a new release, following these instructions...</summary>

1. Fix the bug that caused the failure
2. Start a new `patch` release
3. Skip the **Release Preparation**, and start at the **Release Changes** step
4. Update `CHANGELOG.md` with details about the fix
5. Follow the release checklist for the new Zebra version

</details>

oxarbitrage committed:
chore: Release

teor2345 commented:
CI failed on testnet. It won't block anything merging, but I opened a fix because failures are confusing:
- #7433

oxarbitrage committed:
change estimated release height

oxarbitrage committed:
Merge remote-tracking branch 'origin/main' into bump-v1.2.0

oxarbitrage commented:
I added an estimated release height to be on Monday because there is a PR @teor2345 to include that is not here yet. 

If we end up earlier we could release tomorrow or just keep the Monday. 

Estimated height is now set to `2_215_517`. Coming from:  `2_210_909`(current block)  + `(1152 * 4)`(4 days from now).

change was made at https://github.com/ZcashFoundation/zebra/pull/7431/commits/342a0a22a6b8010402581f28d4e8f156f6a2bc16

oxarbitrage committed:
add 1.2.0 changelog

oxarbitrage commented:
>  If Zebra has started using newer Rust language features or standard library APIs, update the known working Rust version in the README, book, and Cargo.tomls

Zebra will compile with warnings in older rust versions than the last one, so it should be ok to don't upgrade the min version in this release.

teor2345 commented on CHANGELOG.md:
```diff
@@ -5,22 +5,39 @@ All notable changes to Zebra are documented in this file.
 
 - Deduplicate note commitment trees stored in the finalized state ([#7312]()https://github.com/ZcashFoundation/zebra/pull/7312, [#7379](https://github.com/ZcashFoundation/zebra/pull/7379))
 - Insert only the first tree in each series of identical trees into finalized state ([#7266](https://github.com/ZcashFoundation/zebra/pull/7266))
+- Our testing framework now uses the ECC lightwalletd fork ([#7307](https://github.com/ZcashFoundation/zebra/pull/7307)). This was needed to start the work of implementating fast spendability. The ECC repo is now the supported implementation in Zebra, documentation was changed to reflect this. ([#7427](https://github.com/ZcashFoundation/zebra/pull/7427))
```
```suggestion
- Our testing framework now uses the ECC lightwalletd fork ([#7307](https://github.com/ZcashFoundation/zebra/pull/7307)). This was needed to start the work of implementing fast spendability. The ECC repo is now the supported implementation in Zebra, documentation was changed to reflect this. ([#7427](https://github.com/ZcashFoundation/zebra/pull/7427))
```

oxarbitrage replied:
Thanks , fixed in https://github.com/ZcashFoundation/zebra/pull/7431/commits/f11799d0ab8d66dacaae33baf8e0d6d0c307f60c

teor2345 commented on CHANGELOG.md:
```diff
@@ -5,22 +5,39 @@ All notable changes to Zebra are documented in this file.
+
+### Security
+
+- Use the correct state version for databases without a state version file ([#7385](https://github.com/ZcashFoundation/zebra/pull/7385))
```
Just checking why this is a security issue?
(It's a new bug we created in PR #7031 after the last release, and then fixed before this release.)

oxarbitrage replied:
Changed to the fixed section in https://github.com/ZcashFoundation/zebra/pull/7431/commits/f11799d0ab8d66dacaae33baf8e0d6d0c307f60c

teor2345 commented on CHANGELOG.md:
```diff
@@ -5,22 +5,39 @@ All notable changes to Zebra are documented in this file.
--
+- Starts our work implementing "spend before sync" algorithm for lightwalletd.
+- Contains an automatic database upgrade that reduces the size of Zebra's current cached state from approximately 276GB to 244GB. It does so by automatically pruning unneeded note commitment trees from an existing cache. New Zebra instances will also build their cache without these trees.
 
```
We forgot to add this compatibility note to the changelog when we merged #7312.

```suggestion

### Breaking Changes

`zebrad` 1.2.0 cached states are incompatible with previous `zebrad` versions:
- `zebrad` 1.2.0 upgrades the cached state format. The new format is incompatible with previous `zebrad` versions. After upgrading to this Zebra version, don't downgrade to an earlier version.
- When earlier versions try to use states upgraded by `zebrad` 1.2.0:
    - `zebrad` versions 1.0.0 and 1.0.1 will respond to some `z_gettreestate` RPC requests with incorrect empty `final_state` fields
    - pre-release `zebrad` versions can panic when verifying shielded transactions, updating the state, or responding to RPC requests

```

oxarbitrage replied:
Added at https://github.com/ZcashFoundation/zebra/pull/7431/commits/f11799d0ab8d66dacaae33baf8e0d6d0c307f60c

teor2345 commented on CHANGELOG.md:
```diff
@@ -5,22 +5,39 @@ All notable changes to Zebra are documented in this file.
+- Documentation for mining with Docker ([#7179](https://github.com/ZcashFoundation/zebra/pull/7179))
+- Note tree sizes field to `getblock` RPC method ([#7278](https://github.com/ZcashFoundation/zebra/pull/7278))
+- Note commitment subtree types to zebra-chain ([#7371](https://github.com/ZcashFoundation/zebra/pull/7371))
+- Note subtree index handling to zebra-state, but don't write them to the finalized state yet ([#7334](https://github.com/ZcashFoundation/zebra/pull/7334))
```
Users might not know what "them" refers to.

```suggestion
- Note subtree index handling to zebra-state, but we're not writing subtrees to the finalized state yet ([#7334](https://github.com/ZcashFoundation/zebra/pull/7334))
```

oxarbitrage replied:
https://github.com/ZcashFoundation/zebra/pull/7431/commits/f11799d0ab8d66dacaae33baf8e0d6d0c307f60c

teor2345 commented on CHANGELOG.md:
```diff
@@ -5,22 +5,39 @@ All notable changes to Zebra are documented in this file.
+- Note commitment subtree types to zebra-chain ([#7371](https://github.com/ZcashFoundation/zebra/pull/7371))
+- Note subtree index handling to zebra-state, but don't write them to the finalized state yet ([#7334](https://github.com/ZcashFoundation/zebra/pull/7334))
+
+### Fixed
```
This user-visible bug fix seems to be missing:
- #6810 

And this one just got merged:
- #7442

oxarbitrage replied:
Added those to the fixed section in https://github.com/ZcashFoundation/zebra/pull/7431/commits/f11799d0ab8d66dacaae33baf8e0d6d0c307f60c

teor2345 commented on CHANGELOG.md:
```diff
@@ -5,22 +5,39 @@ All notable changes to Zebra are documented in this file.
+
+### Fixed
+
+- Consolidate firebase hosting in production project ([#7313](https://github.com/ZcashFoundation/zebra/pull/7313))
```
Are these CI/CD changes user-visible? (Only user-visible changes need to go in the changelog.)
```suggestion
```

oxarbitrage replied:
https://github.com/ZcashFoundation/zebra/pull/7431/commits/f11799d0ab8d66dacaae33baf8e0d6d0c307f60c

teor2345 commented on CHANGELOG.md:
```diff
@@ -5,22 +5,39 @@ All notable changes to Zebra are documented in this file.
+
+- Consolidate firebase hosting in production project ([#7313](https://github.com/ZcashFoundation/zebra/pull/7313))
+- Log a warning instead of panicking for unused mining configs ([#7290](https://github.com/ZcashFoundation/zebra/pull/7290))
+- Remove all unrequired docker arguments from CI/CD pipelines ([#7231](https://github.com/ZcashFoundation/zebra/pull/7231))
```
```suggestion
```

oxarbitrage replied:
https://github.com/ZcashFoundation/zebra/pull/7431/commits/beacf9af25ef0f4454fe9d92da3aa848b57da6a8

teor2345 reviewed (dismissed):
Thanks! Looks good, just a few tweaks for the changelog.

oxarbitrage committed:
apply suggestions to changelog

Co-authored-by: teor <teor@riseup.net>

oxarbitrage committed:
remove item

oxarbitrage committed:
Merge remote-tracking branch 'origin/main' into bump-v1.2.0

oxarbitrage committed:
change release date to today

oxarbitrage commented:
I changed the release to today as everything is ready https://github.com/ZcashFoundation/zebra/pull/7431/commits/a8b8ce016a3fb9ee57aa5b7ba5d54cf0e86718a6

Right now the tip is at `2_211_804` so i added 576 blocks (about 12 hours) for a final release estimate of `2_212_380`

upbqdn commented:
@oxarbitrage do you have a list of PRs excluded from the changelog?

oxarbitrage commented:
This are the ones from the automatic changelog that were not included.

### Trivial *TODO:* put this in a PR comment, not the CHANGELOG

- Remove custom blank template and enable blank issues ([#7420](https://github.com/ZcashFoundation/zebra/pull/7420))
- Update issue templates to add a custom blank template ([#7367](https://github.com/ZcashFoundation/zebra/pull/7367))
- add(docs): Mention the tree deduplication in CHANGELOG.md ([#7414](https://github.com/ZcashFoundation/zebra/pull/7414))
- build(deps): bump Swatinem/rust-cache from 2.5.1 to 2.6.0 ([#7294](https://github.com/ZcashFoundation/zebra/pull/7294))
- build(deps): bump Swatinem/rust-cache from 2.6.0 to 2.6.1 ([#7341](https://github.com/ZcashFoundation/zebra/pull/7341))
- build(deps): bump Swatinem/rust-cache from 2.6.1 to 2.6.2 ([#7373](https://github.com/ZcashFoundation/zebra/pull/7373))
- build(deps): bump actions/checkout from 3.5.3 to 3.6.0 ([#7425](https://github.com/ZcashFoundation/zebra/pull/7425))
- build(deps): bump r7kamura/rust-problem-matchers from 1.3.0 to 1.4.0 ([#7311](https://github.com/ZcashFoundation/zebra/pull/7311))
- build(deps): bump rustls-webpki from 0.100.1 to 0.100.2 ([#7368](https://github.com/ZcashFoundation/zebra/pull/7368))
- build(deps): bump the app group with 1 update ([#7399](https://github.com/ZcashFoundation/zebra/pull/7399))
- build(deps): bump the app group with 1 update ([#7356](https://github.com/ZcashFoundation/zebra/pull/7356))
- build(deps): bump the app group with 1 update ([#7325](https://github.com/ZcashFoundation/zebra/pull/7325))
- build(deps): bump the app group with 2 updates ([#7282](https://github.com/ZcashFoundation/zebra/pull/7282))
- build(deps): bump the async group with 1 update ([#7323](https://github.com/ZcashFoundation/zebra/pull/7323))
- build(deps): bump the async group with 1 update ([#7300](https://github.com/ZcashFoundation/zebra/pull/7300))
- build(deps): bump the async group with 2 updates ([#7397](https://github.com/ZcashFoundation/zebra/pull/7397))
- build(deps): bump the async group with 2 updates ([#7360](https://github.com/ZcashFoundation/zebra/pull/7360))
- build(deps): bump the data-structures group with 1 update ([#7327](https://github.com/ZcashFoundation/zebra/pull/7327))
- build(deps): bump the formats group with 1 update ([#7283](https://github.com/ZcashFoundation/zebra/pull/7283))
- build(deps): bump the formats group with 2 updates ([#7400](https://github.com/ZcashFoundation/zebra/pull/7400))
- build(deps): bump the formats group with 2 updates ([#7357](https://github.com/ZcashFoundation/zebra/pull/7357))
- build(deps): bump the formats group with 2 updates ([#7336](https://github.com/ZcashFoundation/zebra/pull/7336))
- build(deps): bump the formats group with 4 updates ([#7298](https://github.com/ZcashFoundation/zebra/pull/7298))
- build(deps): bump the log group with 1 update ([#7398](https://github.com/ZcashFoundation/zebra/pull/7398))
- build(deps): bump the log group with 1 update ([#7281](https://github.com/ZcashFoundation/zebra/pull/7281))
- build(deps): bump the log group with 2 updates ([#7355](https://github.com/ZcashFoundation/zebra/pull/7355))
- build(deps): bump the log group with 2 updates ([#7337](https://github.com/ZcashFoundation/zebra/pull/7337))
- build(deps): bump the progress-bar group with 1 update ([#7299](https://github.com/ZcashFoundation/zebra/pull/7299))
- build(deps): bump the test group with 1 update ([#7401](https://github.com/ZcashFoundation/zebra/pull/7401))
- build(deps): bump the test group with 1 update ([#7358](https://github.com/ZcashFoundation/zebra/pull/7358))
- build(deps): bump the test group with 1 update ([#7301](https://github.com/ZcashFoundation/zebra/pull/7301))
- build(deps): bump the test group with 1 update ([#7284](https://github.com/ZcashFoundation/zebra/pull/7284))
- build(deps): bump tj-actions/changed-files from 37.1.2 to 37.3.0 ([#7264](https://github.com/ZcashFoundation/zebra/pull/7264))
- build(deps): bump tj-actions/changed-files from 37.3.0 to 37.4.0 ([#7280](https://github.com/ZcashFoundation/zebra/pull/7280))
- build(deps): bump tj-actions/changed-files from 37.4.0 to 37.5.0 ([#7291](https://github.com/ZcashFoundation/zebra/pull/7291))
- build(deps): bump tj-actions/changed-files from 37.5.0 to 37.5.1 ([#7295](https://github.com/ZcashFoundation/zebra/pull/7295))
- build(deps): bump tj-actions/changed-files from 37.5.1 to 37.6.0 ([#7342](https://github.com/ZcashFoundation/zebra/pull/7342))
- build(deps): bump tj-actions/changed-files from 37.6.0 to 38.0.0 ([#7374](https://github.com/ZcashFoundation/zebra/pull/7374))
- build(deps): bump tj-actions/changed-files from 38.0.0 to 38.1.3 ([#7426](https://github.com/ZcashFoundation/zebra/pull/7426))
- change(checkpoints): Update checkpoints ([#7428](https://github.com/ZcashFoundation/zebra/pull/7428))
- change(ci): Update CI to run tests with zcash/lightwalletd instead of adityapk00/lightwalletd ([#7307](https://github.com/ZcashFoundation/zebra/pull/7307))
- change(deps): Run `cargo update`  ([#7429](https://github.com/ZcashFoundation/zebra/pull/7429))
- change(deps): Use larger dependabot groups ([#7247](https://github.com/ZcashFoundation/zebra/pull/7247))
- cleanup(clippy): Fix a clippy lint and deny.toml entry after recent changes ([#7419](https://github.com/ZcashFoundation/zebra/pull/7419))
- cleanup(clippy): Fix new nightly clippy lints from July and August 2023 ([#7384](https://github.com/ZcashFoundation/zebra/pull/7384))
- cleanup(state): Update some outdated comments in the state \& rpcs ([#7434](https://github.com/ZcashFoundation/zebra/pull/7434))
- fit(whitespace): Update CODEOWNERS ([#7296](https://github.com/ZcashFoundation/zebra/pull/7296))
- fix(build): reintroduce `RUST_LOG` info ([#7404](https://github.com/ZcashFoundation/zebra/pull/7404))
- fix(ci): Show correct lightwalletd cache heights in CI cached state ([#7390](https://github.com/ZcashFoundation/zebra/pull/7390))
- fix(ci): Specify Testnet in jobs that require it ([#7309](https://github.com/ZcashFoundation/zebra/pull/7309))
- fix(ci): Update expected logs in lightwalletd\_update\_sync test ([#7332](https://github.com/ZcashFoundation/zebra/pull/7332))
- fix(ci): Use correct testnet checkpoint job height regex ([#7433](https://github.com/ZcashFoundation/zebra/pull/7433))
- fix(ci): allow to set docs build as a required test ([#7413](https://github.com/ZcashFoundation/zebra/pull/7413))
- fix(clippy): Fix clippy and rustfmt after last releases today ([#7381](https://github.com/ZcashFoundation/zebra/pull/7381))
- fix(doc): consolidate firebase hosting in prod project ([#7313](https://github.com/ZcashFoundation/zebra/pull/7313))
- fix(docs): Simplify the docs for nodes in Sapling trees ([#7378](https://github.com/ZcashFoundation/zebra/pull/7378))
- fix(docs): deploy external and internal docs with the correct path ([#7338](https://github.com/ZcashFoundation/zebra/pull/7338))
- fix(docs): docs after new rust version ([#7375](https://github.com/ZcashFoundation/zebra/pull/7375))
- fix(docs): redundant links in state ([#7402](https://github.com/ZcashFoundation/zebra/pull/7402))
- fix(panic): Log a warning instead of panicking for unused mining configs ([#7290](https://github.com/ZcashFoundation/zebra/pull/7290))
- fix(state): Avoid temporary failures verifying the first non-finalized block or attempting to fork the chain before the final checkpoint ([#6810](https://github.com/ZcashFoundation/zebra/pull/6810))
- fix(test): Update expected log in lightwalletd\_full\_sync test ([#7315](https://github.com/ZcashFoundation/zebra/pull/7315))
- ref(docker): remove all unrequired docker arguments from CI/CD pipelines ([#7231](https://github.com/ZcashFoundation/zebra/pull/7231))
- ref(docs): consolidate docs linting and publishing ([#7339](https://github.com/ZcashFoundation/zebra/pull/7339))

then a few more that were duplicated were removed plus the ones in @teor2345 review.

upbqdn reviewed (approved):
All looks good.

oxarbitrage commented:
Output of installation from `crates.io`:

```
oxarbitrage@oxarbitrage:/media/oxarbitrage/4eb53770-4e4f-4d80-a830-0914f3f5b89a/zebra/1.2.0_public_crates/zebra$ cargo install --locked --force --version 1.2.0 zebrad && ~/.cargo/bin/zebrad
  Downloaded zebrad v1.2.0
  Downloaded 1 crate (234.7 KB) in 2.29s
    Updating crates.io index
  Installing zebrad v1.2.0
    Updating crates.io index
    Updating crates.io index
   Compiling libc v0.2.147
   Compiling proc-macro2 v1.0.66
   Compiling unicode-ident v1.0.11
   Compiling cfg-if v1.0.0
   Compiling version_check v0.9.4
   Compiling autocfg v1.1.0
   Compiling once_cell v1.18.0
   Compiling serde v1.0.188
   Compiling subtle v2.4.1
   Compiling typenum v1.16.0
   Compiling log v0.4.20
   Compiling pin-project-lite v0.2.13
   Compiling spin v0.5.2
   Compiling memchr v2.6.2
   Compiling arrayvec v0.7.4
   Compiling ppv-lite86 v0.2.17
   Compiling lazy_static v1.4.0
   Compiling tracing-core v0.1.31
   Compiling byteorder v1.4.3
   Compiling radium v0.7.0
   Compiling generic-array v0.14.7
   Compiling crossbeam-utils v0.8.16
   Compiling tap v1.0.1
   Compiling nonempty v0.7.0
   Compiling wyz v0.5.1
   Compiling scopeguard v1.2.0
   Compiling cpufeatures v0.2.9
   Compiling funty v2.0.0
   Compiling futures-core v0.3.28
   Compiling arrayref v0.3.7
   Compiling constant_time_eq v0.2.6
   Compiling syn v1.0.109
   Compiling memoffset v0.9.0
   Compiling memuse v0.2.1
   Compiling quote v1.0.33
   Compiling num-traits v0.2.16
   Compiling bitvec v1.0.1
   Compiling jobserver v0.1.26
   Compiling syn v2.0.29
   Compiling getrandom v0.2.10
   Compiling num_cpus v1.16.0
   Compiling crossbeam-epoch v0.9.15
   Compiling cc v1.0.83
   Compiling rand_core v0.6.4
   Compiling futures-sink v0.3.28
   Compiling rand_chacha v0.3.1
   Compiling blake2b_simd v1.0.1
   Compiling ahash v0.8.3
   Compiling either v1.9.0
   Compiling rand v0.8.5
   Compiling static_assertions v1.1.0
   Compiling crypto-common v0.1.6
   Compiling block-buffer v0.10.4
   Compiling crossbeam-channel v0.5.8
   Compiling digest v0.10.7
   Compiling slab v0.4.9
   Compiling futures-channel v0.3.28
   Compiling bytes v1.4.0
   Compiling rayon-core v1.11.0
   Compiling futures-task v0.3.28
   Compiling crossbeam-deque v0.8.3
   Compiling inout v0.1.3
   Compiling futures-util v0.3.28
   Compiling semver v1.0.18
   Compiling ring v0.16.20
   Compiling glob v0.3.1
   Compiling mio v0.8.8
   Compiling signal-hook-registry v1.4.1
   Compiling socket2 v0.5.3
   Compiling pin-utils v0.1.0
   Compiling futures-io v0.3.28
   Compiling clang-sys v1.6.1
   Compiling untrusted v0.7.1
   Compiling crunchy v0.2.2
   Compiling prettyplease v0.2.12
   Compiling tinyvec_macros v0.1.1
   Compiling tinyvec v1.6.0
   Compiling rayon v1.7.0
   Compiling ff v0.13.0
   Compiling group v0.13.0
   Compiling aho-corasick v1.0.5
   Compiling pairing v0.23.0
   Compiling allocator-api2 v0.2.16
   Compiling regex-syntax v0.7.5
   Compiling thiserror v1.0.47
   Compiling itoa v1.0.9
   Compiling unicode-normalization v0.1.22
   Compiling pasta_curves v0.5.1
   Compiling rustc_version v0.4.0
   Compiling hashbrown v0.14.0
   Compiling num-integer v0.1.45
   Compiling minimal-lexical v0.2.1
   Compiling libloading v0.7.4
   Compiling bls12_381 v0.8.0
   Compiling secp256k1-sys v0.8.1
   Compiling nom v7.1.3
   Compiling sha2 v0.10.7
   Compiling universal-hash v0.5.1
   Compiling num-bigint v0.4.4
   Compiling regex-automata v0.3.7
   Compiling equivalent v1.0.1
   Compiling opaque-debug v0.3.0
   Compiling ident_case v1.0.1
   Compiling libm v0.2.7
   Compiling cexpr v0.6.0
   Compiling poly1305 v0.8.0
   Compiling serde_derive v1.0.188
   Compiling zeroize_derive v1.4.2
   Compiling tracing-attributes v0.1.26
   Compiling tokio-macros v2.1.0
   Compiling futures-macro v0.3.28
   Compiling zeroize v1.6.0
   Compiling cipher v0.4.4
   Compiling tokio v1.32.0
   Compiling thiserror-impl v1.0.47
   Compiling regex v1.9.4
   Compiling tracing v0.1.37
   Compiling chacha20 v0.9.1
   Compiling maybe-rayon v0.1.1
   Compiling jubjub v0.10.0
   Compiling aead v0.5.2
   Compiling rustc-hash v1.1.0
   Compiling peeking_take_while v0.1.2
   Compiling base64ct v1.0.1
   Compiling halo2_legacy_pdqsort v0.1.0
   Compiling lazycell v1.3.0
   Compiling shlex v1.1.0
   Compiling password-hash v0.3.2
   Compiling halo2_proofs v0.3.0
   Compiling chacha20poly1305 v0.10.1
   Compiling cbc v0.1.2
   Compiling incrementalmerkletree v0.4.0
   Compiling futures-executor v0.3.28
   Compiling platforms v3.1.2
   Compiling strsim v0.10.0
   Compiling fnv v1.0.7
   Compiling bitflags v1.3.2
   Compiling serde_json v1.0.105
   Compiling darling_core v0.20.3
   Compiling curve25519-dalek v4.0.0
   Compiling futures v0.3.28
   Compiling pbkdf2 v0.10.1
   Compiling zcash_note_encryption v0.4.0
   Compiling aes v0.8.3
   Compiling fpe v0.6.1
   Compiling impl-trait-for-tuples v0.2.2
   Compiling bs58 v0.5.0
   Compiling hmac v0.12.1
   Compiling f4jumble v0.1.0
   Compiling blake2s_simd v1.0.1
   Compiling zcash_encoding v0.2.0
   Compiling bech32 v0.9.1
   Compiling byte-slice-cast v1.2.2
   Compiling ryu v1.0.15
   Compiling zcash_address v0.3.0
   Compiling bip0039 v0.10.1
   Compiling tokio-util v0.7.8
   Compiling hex v0.4.3
   Compiling indexmap v2.0.0
   Compiling uint v0.9.5
   Compiling reddsa v0.5.1
   Compiling darling_macro v0.20.3
   Compiling curve25519-dalek-derive v0.1.0
   Compiling ripemd v0.1.3
   Compiling halo2_gadgets v0.3.0
   Compiling equihash v0.2.0
   Compiling rustc-hex v2.1.0
   Compiling bitflags v2.4.0
   Compiling pkg-config v0.3.27
   Compiling darling v0.20.3
   Compiling fixed-hash v0.7.0
   Compiling redjubjub v0.7.0
   Compiling pin-project-internal v1.1.3
   Compiling link-cplusplus v1.0.9
   Compiling tower-service v0.3.2
   Compiling getrandom v0.1.16
   Compiling signature v2.1.0
   Compiling smallvec v1.11.0
   Compiling bindgen v0.65.1
   Compiling vcpkg v0.2.15
   Compiling ed25519 v2.2.2
   Compiling serde_with_macros v3.3.0
   Compiling bridgetree v0.3.0
   Compiling webpki v0.22.0
   Compiling lz4-sys v1.9.4
   Compiling parity-scale-codec v3.6.5
   Compiling libz-sys v1.1.12
   Compiling cxxbridge-flags v1.0.95
   Compiling regex-syntax v0.6.29
   Compiling pin-project v1.1.3
   Compiling termcolor v1.2.0
   Compiling unicode-width v0.1.10
   Compiling iana-time-zone v0.1.57
   Compiling orchard v0.5.0
   Compiling impl-codec v0.6.0
   Compiling primitive-types v0.11.1
   Compiling bindgen v0.66.1
   Compiling humantime v2.1.0
   Compiling rustls v0.20.9
   Compiling which v4.4.0
   Compiling serde_with v3.3.0
   Compiling codespan-reporting v0.11.1
   Compiling chrono v0.4.27
   Compiling cxx v1.0.95
   Compiling zcash_history v0.3.0
   Compiling regex-automata v0.1.10
   Compiling ed25519-zebra v4.0.2
   Compiling x25519-dalek v2.0.0
   Compiling bitflags-serde-legacy v0.1.1
   Compiling serde-big-array v0.5.1
   Compiling sct v0.7.0
   Compiling metrics-macros v0.7.0
   Compiling displaydoc v0.2.4
   Compiling hdrhistogram v7.5.2
   Compiling itertools v0.11.0
   Compiling rustversion v1.0.14
   Compiling minreq v2.8.1
   Compiling overload v0.1.1
   Compiling tower-layer v0.3.2
   Compiling rustix v0.38.10
   Compiling tower v0.4.13
   Compiling nu-ansi-term v0.46.0
   Compiling metrics v0.21.1
   Compiling rand_core v0.5.1
   Compiling matchers v0.1.0
   Compiling cxx-gen v0.7.107
   Compiling webpki-roots v0.22.6
   Compiling libgit2-sys v0.15.2+1.6.4
   Compiling tracing-log v0.1.3
   Compiling sharded-slab v0.1.4
   Compiling thread_local v1.1.7
   Compiling linux-raw-sys v0.4.5
   Compiling option-ext v0.2.0
   Compiling percent-encoding v2.3.0
   Compiling unicode-bidi v0.3.13
   Compiling dirs-sys v0.4.1
   Compiling form_urlencoded v1.2.0
   Compiling tracing-subscriber v0.3.17
   Compiling idna v0.4.0
   Compiling librocksdb-sys v0.11.0+8.1.1
   Compiling rand_chacha v0.2.2
   Compiling bellman v0.14.0
   Compiling cxxbridge-macro v1.0.95
   Compiling lock_api v0.4.10
   Compiling unicase v2.7.0
   Compiling parking_lot_core v0.8.6
   Compiling rlimit v0.10.1
   Compiling known-folders v1.0.1
   Compiling utf8parse v0.2.1
   Compiling xdg v2.5.2
   Compiling httparse v1.8.0
   Compiling fastrand v2.0.0
   Compiling home v0.5.5
   Compiling anyhow v1.0.75
   Compiling tempfile v3.8.0
   Compiling anstyle-parse v0.2.1
   Compiling vergen v8.2.4
   Compiling rand v0.7.3
   Compiling url v2.4.1
   Compiling dirs v5.0.1
   Compiling http v0.2.9
   Compiling tracing-futures v0.2.5
   Compiling tokio-stream v0.1.14
   Compiling env_logger v0.7.1
   Compiling backtrace v0.3.69
   Compiling zcash_script v0.1.13
   Compiling bstr v1.6.2
   Compiling instant v0.1.12
   Compiling time-core v0.1.1
   Compiling eyre v0.6.8
   Compiling anstyle v1.0.2
   Compiling try-lock v0.2.4
   Compiling anstyle-query v1.0.0
   Compiling gimli v0.28.0
   Compiling adler v1.0.2
   Compiling colorchoice v1.0.0
   Compiling num_threads v0.1.6
   Compiling deranged v0.3.8
   Compiling anstream v0.5.0
   Compiling miniz_oxide v0.7.1
   Compiling time v0.3.28
   Compiling toml v0.5.11
   Compiling want v0.3.1
   Compiling globset v0.4.13
   Compiling http-body v0.4.5
   Compiling addr2line v0.21.0
   Compiling quickcheck v0.9.2
   Compiling jsonrpc-core v18.0.0
   Compiling bincode v1.3.3
   Compiling tokio-util v0.6.10
   Compiling quickcheck_macros v0.9.1
   Compiling socket2 v0.4.9
   Compiling object v0.32.0
   Compiling cfg-if v0.1.10
   Compiling indenter v0.3.3
   Compiling mset v0.1.1
   Compiling httpdate v1.0.3
   Compiling clap_lex v0.5.1
   Compiling heck v0.4.1
   Compiling rustc-demangle v0.1.23
   Compiling unicode-xid v0.2.4
   Compiling clap_derive v4.4.0
   Compiling ordered-map v0.4.2
   Compiling synstructure v0.12.6
   Compiling clap_builder v4.4.1
   Compiling hyper v0.14.27
   Compiling net2 v0.2.39
   Compiling jsonrpc-server-utils v18.0.0
   Compiling parking_lot v0.11.2
   Compiling proc-macro-crate v0.1.5
   Compiling tower-batch-control v0.2.41-beta.5
   Compiling tracing-error v0.2.0
   Compiling tower-fallback v0.2.41-beta.5
   Compiling humantime-serde v1.1.1
   Compiling toml_datetime v0.6.3
   Compiling serde_spanned v0.6.3
   Compiling owo-colors v3.5.0
   Compiling winnow v0.5.15
   Compiling clap v4.4.1
   Compiling jsonrpc-http-server v18.0.0
   Compiling jsonrpc-derive v18.0.0
   Compiling color-eyre v0.6.2
   Compiling abscissa_derive v0.7.0
   Compiling toml_edit v0.19.14
   Compiling secrecy v0.8.0
   Compiling wait-timeout v0.2.0
   Compiling canonical-path v2.0.2
   Compiling fs-err v2.9.0
   Compiling arc-swap v1.6.0
   Compiling tracing-appender v0.2.2
   Compiling atty v0.2.14
   Compiling abscissa_core v0.7.0
   Compiling toml v0.7.6
   Compiling secp256k1 v0.26.0
   Compiling hdwallet v0.4.1
   Compiling zcash_primitives v0.12.0
   Compiling zebra-chain v1.0.0-beta.29
   Compiling zcash_proofs v0.12.1
   Compiling zebra-node-services v1.0.0-beta.29
   Compiling zebra-network v1.0.0-beta.29
   Compiling zebra-script v1.0.0-beta.29
   Compiling git2 v0.17.2
   Compiling zebrad v1.2.0
   Compiling rocksdb v0.21.0
   Compiling zebra-state v1.0.0-beta.29
   Compiling zebra-consensus v1.0.0-beta.29
   Compiling zebra-rpc v1.0.0-beta.29
    Finished release [optimized] target(s) in 5m 05s
   Replacing /home/oxarbitrage/.cargo/bin/zebrad
    Replaced package `zebrad v1.0.0-rc.9 (https://github.com/ZcashFoundation/zebra?branch=es-serialize-buld-error#29ff6512)` with `zebrad v1.2.0` (executable `zebrad`)























































              X@8:::::::@X                      S888@t             S888%        
          @::X:;SX;%8S8@@:X::8              S:         .8      @.         .@    
       @:;@8@.S8  ;;t;. XX.;@@;:@         ;               %  ;               X  
     8:@88     .%@.S@.XS:     888:8      8                 t%                 @ 
    :;SX    XS8;::@X@8::;8t8:   8;;:@                                           
  @::@    88:::8 8::88::::::X@    8::8  @                                      8
 @:XS8   ;::::::8.88%8@S88::::8:  8S8:8 X                                      X
 ::;8 t8S8::::@St 88SX @SS@8:8S;8S @@:: 8                                      8
8:XX  %%@@::X% 8:  X@ t:8t X ..%t8  ;8:8                                        
::8t:  tX:X8tS 8: :@ ;.8@.X tt888:  @8:: X                                    % 
::8S:  @:8Xtt% 8: S 8 8S.   .8;::.  88::  %                                  t  
::X8   S@8   8:8.@8@8Xt S  .%8;S    %8::   8                                8   
 ::@8  S .   @:::::::X;@ 8.88S8    @8;:X     %                            %     
 8:X8X .@:X88:::::::::888 @;:X    %88::        %                        %       
  @::S    888@8:::::::::8X; 8     8::8           8                    8         
   X:;S@   tXX8;::::::::S::@8   8;S:8              S.              :t           
     ::8S@     :S88XX@8X:     @XX::                   %          %              
       8:;%X8.:S        t; X8%S:8X                      @      S                
         X8::8SS8SX8@@X@8;8S:8X                           S  %                  
              X@8:::::::@XS                                %                    

Thank you for running a mainnet zebrad 1.2.0 node!
You're helping to strengthen the network and contributing to a social good :)
2023-09-02T15:27:36.442068Z  INFO zebrad::components::tracing::component: started tracing component filter="info" TRACING_STATIC_MAX_LEVEL=LevelFilter::INFO LOG_STATIC_MAX_LEVEL=Info
2023-09-02T15:27:36.442084Z  INFO zebrad::application: Diagnostic Metadata:
version: 1.2.0
Zcash network: Mainnet
running state version: 25.1.1
initial disk state version: 25.0.2
features: default,release_max_level_info
target triple: x86_64-unknown-linux-gnu
rust compiler: 1.72.0
rust release date: 2023-08-23
optimization level: 3
debug checks: false
2023-09-02T15:27:36.442090Z  INFO zebrad::application: loaded zebrad config config_path=None config=ZebradConfig { consensus: Config { checkpoint_sync: true, debug_skip_parameter_preload: false }, metrics: Config { endpoint_addr: None }, network: Config { listen_addr: 0.0.0.0:8233, network: Mainnet, initial_mainnet_peers: {"dnsseed.z.cash:8233", "dnsseed.str4d.xyz:8233", "mainnet.seeder.zfnd.org:8233", "mainnet.is.yolo.money:8233"}, initial_testnet_peers: {"dnsseed.testnet.z.cash:18233", "testnet.seeder.zfnd.org:18233", "testnet.is.yolo.money:18233"}, cache_dir: IsEnabled(true), peerset_initial_target_size: 25, crawl_new_peer_interval: 61s, max_connections_per_ip: 1 }, state: Config { cache_dir: "/home/oxarbitrage/.cache/zebra", ephemeral: false, debug_stop_at_height: None, delete_old_database: true }, tracing: Config { use_color: true, force_use_color: false, filter: None, buffer_limit: 128000, endpoint_addr: None, flamegraph: None, log_file: None, use_journald: false }, sync: Config { download_concurrency_limit: 50, checkpoint_verify_concurrency_limit: 1000, full_verify_concurrency_limit: 20, parallel_cpu_threads: 0 }, mempool: Config { tx_cost_limit: 80000000, eviction_memory_time: 3600s, debug_enable_at_height: None }, rpc: Config { listen_addr: None, parallel_cpu_threads: 1, debug_force_finished_sync: false }, mining: Config { miner_address: None, extra_coinbase_data: None, debug_like_zcashd: true } }
2023-09-02T15:27:36.442131Z  INFO {net="Main"}: zebrad::application: initialized rayon thread pool for CPU-bound tasks num_threads=12
2023-09-02T15:27:36.442381Z  INFO {net="Main"}: zebrad::commands::start: Starting zebrad
2023-09-02T15:27:36.442407Z  INFO {net="Main"}: zebrad::commands::start: initializing node state
2023-09-02T15:27:36.446660Z  INFO {net="Main"}: zebrad::commands::start: opening database, this may take a few minutes
2023-09-02T15:27:36.446724Z  INFO zebra_state::service::finalized_state::disk_format::upgrade: trying to open older database format: launching upgrade task running_version=Version { major: 25, minor: 1, patch: 1 } disk_version=Version { major: 25, minor: 0, patch: 2 }
2023-09-02T15:27:36.446778Z  INFO zebra_state::service::finalized_state::disk_db: the open file limit is high enough for Zebra current_limit=1024 min_limit=512 ideal_limit=1024
2023-09-02T15:27:36.531064Z  INFO zebra_state::service::finalized_state::disk_db: Opened Zebra state cache at /home/oxarbitrage/.cache/zebra/state/v25/mainnet
2023-09-02T15:27:36.532906Z  INFO zebra_state::service::finalized_state: loaded Zebra state cache tip=Some((Height(2800), block::Hash("00000001ecd754790237618cb79c4cd302e52571ecda7a80e6113c5e423c0e55")))
2023-09-02T15:27:36.533142Z  INFO zebra_state::service: starting legacy chain check
2023-09-02T15:27:36.537634Z  INFO zebra_state::service: cached state consensus branch is valid: no legacy chain found
2023-09-02T15:27:36.537743Z  INFO {net="Main"}: zebrad::commands::start: initializing network
2023-09-02T15:27:36.537787Z  INFO {net="Main"}:open_listener{addr=0.0.0.0:8233}: zebra_network::peer_set::initialize: Trying to open Zcash protocol endpoint at 0.0.0.0:8233...
2023-09-02T15:27:36.537827Z  INFO {net="Main"}:open_listener{addr=0.0.0.0:8233}: zebra_network::peer_set::initialize: Opened Zcash protocol endpoint at 0.0.0.0:8233
2023-09-02T15:27:36.537866Z  INFO {net="Main"}: zebra_network::address_book_updater: starting the address book updater
2023-09-02T15:27:36.539235Z  INFO zebra_state::service::finalized_state::disk_format::upgrade: marked database format as upgraded running_version=Version { major: 25, minor: 1, patch: 1 } format_upgrade_version=Version { major: 25, minor: 1, patch: 1 } disk_version=Version { major: 25, minor: 0, patch: 2 }
2023-09-02T15:27:36.539253Z  INFO zebra_state::service::finalized_state::disk_format::upgrade: Zebra automatically upgraded the database format to: newer_running_version=Version { major: 25, minor: 1, patch: 1 }
2023-09-02T15:27:37.264324Z  INFO {net="Main"}:add_initial_peers: zebra_network::config: resolved seed peer IP addresses seed="dnsseed.str4d.xyz:8233" remote_ip_count=28
2023-09-02T15:27:37.588072Z  INFO {net="Main"}:add_initial_peers: zebra_network::config: resolved seed peer IP addresses seed="dnsseed.z.cash:8233" remote_ip_count=25
2023-09-02T15:27:38.177971Z  INFO {net="Main"}:add_initial_peers: zebra_network::config: resolved seed peer IP addresses seed="mainnet.seeder.zfnd.org:8233" remote_ip_count=25
2023-09-02T15:27:41.538400Z  INFO {net="Main"}:add_initial_peers: zebra_network::config: DNS timeout resolving peer IP addresses host="mainnet.is.yolo.money:8233" e=Elapsed(())
2023-09-02T15:27:41.538438Z  INFO {net="Main"}:add_initial_peers: zebra_network::config: Seed peer DNS resolution failed, checking for addresses from other seed peers host="mainnet.is.yolo.money:8233" attempts=1
2023-09-02T15:27:41.539851Z  INFO {net="Main"}:add_initial_peers: zebra_network::config: loaded cached peer IP addresses cached_ip_count=29 peer_cache_file="/home/oxarbitrage/.cache/zebra/network/mainnet.peers"
2023-09-02T15:27:41.539897Z  INFO {net="Main"}:add_initial_peers: zebra_network::peer_set::initialize: limiting the initial peers list from 75 to 25
2023-09-02T15:27:41.539961Z  INFO {net="Main"}:add_initial_peers: zebra_network::peer_set::initialize: connecting to initial peer set ipv4_peer_count=25 ipv6_peer_count=0
2023-09-02T15:27:42.405211Z  INFO {net="Main"}:add_initial_peers: zebra_network::peer_set::initialize: an initial peer connection failed successes=3 errors=1 addr=v4redacted:8233 e=Connection refused (os error 111)
2023-09-02T15:27:44.353230Z  INFO {net="Main"}:add_initial_peers: zebra_network::peer_set::initialize: finished connecting to initial seed and disk cache peers handshake_success_total=24 handshake_error_total=1 outbound_connections=24
2023-09-02T15:27:44.353301Z  INFO {net="Main"}: zebra_network::peer_set::initialize: sending initial request for peers active_initial_peer_count=24
2023-09-02T15:27:44.353679Z  INFO {net="Main"}: zebrad::commands::start: initializing verifiers
2023-09-02T15:27:44.353728Z  INFO {net="Main"}:crawl_and_dial{new_peer_interval=61s}: zebra_network::peer_set::initialize: starting the peer address crawler crawl_new_peer_interval=61s outbound_connections=24
2023-09-02T15:27:44.353767Z  INFO {net="Main"}:init{config=Config { checkpoint_sync: true, debug_skip_parameter_preload: false } network=Mainnet debug_skip_parameter_preload=false}: zebra_consensus::router: starting state checkpoint validation
2023-09-02T15:27:44.353953Z  INFO {net="Main"}:init{config=Config { checkpoint_sync: true, debug_skip_parameter_preload: false } network=Mainnet debug_skip_parameter_preload=false}: zebra_consensus::primitives::groth16::params: checking and loading Zcash Sapling and Sprout parameters
2023-09-02T15:27:44.360818Z  INFO {net="Main"}:init{config=Config { checkpoint_sync: true, debug_skip_parameter_preload: false } network=Mainnet debug_skip_parameter_preload=false}: zebra_consensus::router: initializing block verifier router tip=Some((Height(2800), block::Hash("00000001ecd754790237618cb79c4cd302e52571ecda7a80e6113c5e423c0e55"))) max_checkpoint_height=Height(2208698)
2023-09-02T15:27:44.360932Z  INFO {net="Main"}: zebrad::commands::start: initializing syncer
2023-09-02T15:27:44.360957Z  INFO {net="Main"}: zebrad::commands::start: initializing mempool
2023-09-02T15:27:44.360970Z  INFO {net="Main"}: zebrad::commands::start: fully initializing inbound peer request handler
2023-09-02T15:27:44.361020Z  INFO {net="Main"}: zebrad::components::sync::gossip: initializing block gossip task
2023-09-02T15:27:44.361025Z  INFO {net="Main"}: zebra_state::config: checking for old database versions
2023-09-02T15:27:44.361021Z  INFO {net="Main"}: zebrad::components::mempool::queue_checker: initializing mempool queue checker task
2023-09-02T15:27:44.361029Z  INFO {net="Main"}: zebrad::commands::start: spawned initial Zebra tasks
2023-09-02T15:27:44.361041Z  INFO {net="Main"}: zebrad::components::mempool::gossip: initializing transaction gossip task
2023-09-02T15:27:44.361053Z  INFO {net="Main"}: zebrad::components::sync::end_of_support: Starting end of support task
2023-09-02T15:27:44.361066Z  INFO {net="Main"}: zebrad::components::mempool::crawler: initializing mempool crawler task
2023-09-02T15:27:44.361077Z  INFO {net="Main"}: zebra_state::config: finished old database version cleanup task
2023-09-02T15:27:44.361573Z  INFO {net="Main"}:init{config=Config { checkpoint_sync: true, debug_skip_parameter_preload: false } network=Mainnet debug_skip_parameter_preload=false}: zebra_consensus::router: state is not fully synced yet, remaining checkpoints will be verified during syncing
2023-09-02T15:27:44.361588Z  INFO {net="Main"}:init{config=Config { checkpoint_sync: true, debug_skip_parameter_preload: false } network=Mainnet debug_skip_parameter_preload=false}: zebra_consensus::router: finished state checkpoint validation
2023-09-02T15:27:44.361932Z  INFO {net="Main"}:sync:try_to_sync: zebrad::components::sync: starting sync, obtaining new tips state_tip=Some(Height(2800))
2023-09-02T15:27:48.344007Z  INFO {net="Main"}:init{config=Config { checkpoint_sync: true, debug_skip_parameter_preload: false } network=Mainnet debug_skip_parameter_preload=false}: zebra_consensus::primitives::groth16::params: Zcash Sapling and Sprout parameters downloaded and/or verified
2023-09-02T15:27:49.150856Z  INFO {net="Main"}:sync:try_to_sync: zebrad::components::sync: extending tips tips.len=1 in_flight=499 extra_hashes=0 lookahead_limit=1000 state_tip=Some(Height(2800))
2023-09-02T15:27:54.361366Z  INFO {net="Main"}: zebrad::components::sync::end_of_support: Checking if Zebra release is inside support range ...
2023-09-02T15:27:54.361409Z  INFO {net="Main"}: zebrad::components::sync::end_of_support: Zebra release is supported until block 2276892, please report bugs at https://github.com/ZcashFoundation/zebra/issues
2023-09-02T15:27:59.326805Z  INFO {net="Main"}:sync:try_to_sync: zebrad::components::sync: extending tips tips.len=1 in_flight=997 extra_hashes=0 lookahead_limit=1000 state_tip=Some(Height(2800))
2023-09-02T15:28:04.355274Z  INFO {net="Main"}:peer_cache_updater: zebra_network::config: updated cached peer IP addresses cached_ip_count=27 peer_cache_file="/home/oxarbitrage/.cache/zebra/network/mainnet.peers"
^C2023-09-02T15:28:04.886027Z  INFO {net="Main"}:sig{kind=SignalKind(2) name="SIGINT"}: zebrad::signal: received SIGINT, starting shutdown
2023-09-02T15:28:04.886095Z  INFO {net="Main"}: zebrad::components::tokio: waiting for async tokio tasks to shut down TOKIO_SHUTDOWN_TIMEOUT=20s
2023-09-02T15:28:04.886213Z  INFO zebra_state::service::write: StateService closed the block reset channel. Is Zebra shutting down?
2023-09-02T15:28:04.886350Z  INFO zebra_state::service::finalized_state::disk_db: forcing shutdown of a state database with multiple active instances path="/home/oxarbitrage/.cache/zebra/state/v25/mainnet"
2023-09-02T15:28:04.886734Z  INFO {net="Main"}: zebra_network::address_book_updater: stopping address book updater error=Err(AllAddressBookUpdaterSendersClosed)
2023-09-02T15:28:04.888612Z  INFO zebra_state::service: waiting for the block write task to finish
2023-09-02T15:28:04.946346Z  INFO {net="Main"}: zebrad::components::tokio: shutting down Zebra
2023-09-02T15:28:04.946362Z  INFO {net="Main"}: zebrad::commands::start: stopping zebrad
oxarbitrage@oxarbitrage:/media/oxarbitrage/4eb53770-4e4f-4d80-a830-0914f3f5b89a/zebra/1.2.0_public_crates/zebra$ 
```
